### PR TITLE
docs: update release process with recent improvements

### PR DESCRIPTION
**Pull request by orndorffgrant**
## Proposed Commit Message

```
docs: update release process with recent improvements
```

## Test Steps

Read the updates and verify that they reflect reality and make sense.

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [x] I have updated or added any documentation accordingly

### CODE REVIEW

**Review by renanrodrigo — APPROVED** [review]

File: RELEASES.md [hunk 1]
```diff
@@ -183,16 +183,15 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
 
     a. Ask in ~Server for a review of your MPs. Include a link to the primary MP into ubuntu/devel and mention the other MPs are only changelog MPs for the SRUs into past releases.
     
-    b. If they request changes, create a PR into `main` on github and ask UAClient team for review. After a merge, open another PR on github, cherry-picking the change into the release branch. After that is merged, cherry-pick the commit into your `upload-<this-version>-impish` branch and push to launchpad. Then notify the Server Team member that you have addressed their requests. (This can probably be simplified).
+    b. If they request changes, create a PR into the release branch on github and ask UAClient team for review. After that is merged, cherry-pick the commit into your `upload-<this-version>-impish` branch and push to launchpad. You'll also need to rebase the other `upload-<this-version>-<series>` branches and force push them to launchpad. Then notify the Server Team member that you have addressed their requests.
```
Comment (lucasmoura) [inline RELEASES.md:186]: Maybe we should replace `impish` here for `<latest-release>`

File: RELEASES.md [hunk 2]
```diff
@@ -240,7 +239,13 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
...
     h. Once all SRU bugs are tagged as `verification*-done`, all SRU-bugs should be listed as green in [the pending_sru page](https://people.canonical.com/~ubuntu-archive/pending-sru.html).
     
-    i. After the pending_sru page says that ubuntu-advantage-tools has been in proposed for 7 days, it is now time to ping the [current SRU vanguard](https://wiki.ubuntu.com/StableReleaseUpdates#Publishing) for acceptance of ubuntu-advantage-tools into -updates.
+    i. After the pending sru page says that ubuntu-advantage-tools has been in proposed for 7 days, it is now time to ping the [current SRU vanguard](https://wiki.ubuntu.com/StableReleaseUpdates#Publishing) for acceptance of ubuntu-advantage-tools into -updates.
+
+    j. Ping the Ubuntu Server team member who approved the version in step `II.4` to now upload to the devel release.
+
+    k. Check `rmadison ubuntu-advantage-tools` for updated version in devel release
+
+    l. Confirm availability in <devel>-updates pocket via `lxc launch ubuntu-daily:impish dev-i; lxc exec dev-i -- apt update; lxc exec dev-i -- apt-cache policy ubuntu-advantage-tools`
```
Comment (lucasmoura) [inline RELEASES.md:248]: Same thing here

**Comment by orndorffgrant** [comment]
Good catch @lucasmoura! Should be updated now. I changed it to `<devel-series>` to emphasize that it is the release currently in development.

File: RELEASES.md [hunk 3]
```diff
@@ -183,16 +183,15 @@ If this is your first time releasing ubuntu-advantage-tools, you'll need to do t
...
     c. Once review is complete and approved, confirm that Ubuntu Server approver will be tagging the PR with the appropriate `upload/<version>` tag so git-ubuntu will import rich commit history.
 
-    d. Check `rmadison ubuntu-advantage-tools` for updated version in devel release
-
-    e. Confirm availability in <devel>-updates pocket via `lxc launch ubuntu-daily:impish dev-i; lxc exec dev-i -- apt update; lxc exec dev-i -- apt-cache policy ubuntu-advantage-tools`
-      * Note that any changes to the code after this point will likely require a bump in the patch version of the release.
+    d. At this point the Server Team member should **not** upload the version to the devel release.
+      * If they do, then any changes to the code after this point will require a bump in the patch version of the release.
 
     f. Ask Ubuntu Server approver if they also have upload rights to the proposed queue. If they do, request that they upload ubuntu-advantage-tools for all releases. If they do not, ask in ~Server channel for a Ubuntu Server team member with upload rights for an upload review of the MP for the proposed queue.
```
Comment (blackboxsw) [inline RELEASES.md:196]: Minor fix needed  for "f" "g" "h" -> "e" "f" "g"
```suggestion
    e. Ask Ubuntu Server approver if they also have upload rights to the proposed queue. If they do, request that they upload ubuntu-advantage-tools for all releases. If they do not, ask in ~Server channel for a Ubuntu Server team member with upload rights for an upload review of the MP for the proposed queue.
```

Comment (orndorffgrant) [reply RELEASES.md:196]: Ah good catch!

Comment (orndorffgrant) [reply RELEASES.md]: @blackboxsw updated!

**Review by blackboxsw — APPROVED** [review]
nice updates. +1

**Pull request merged**